src/components/intercom/index.tsx

```diff
@@ -84,23 +84,27 @@ export const Intercom: FC<Props> = ({
   autoload = false,
   userInfo = {},
 }) => {
-  const intercomSettings = {
-    cfy: true,
-    app_id: appId,
-    stage,
-    custom_launcher_selector: `#${intercomLauncherId}`,
-    hide_default_launcher: true,
-    alignment: "right",
-    horizontal_padding: 20,
-    vertical_padding: 76,
-    language_override: language,
-    ...userInfo,
-  }
+  const intercomSettings = useMemo(() => {
+    return {
+      cfy: true,
+      app_id: appId,
+      stage,
+      custom_launcher_selector: `#${intercomLauncherId}`,
+      hide_default_launcher: true,
+      alignment: "right",
+      horizontal_padding: 20,
+      vertical_padding: 76,
+      language_override: language,
+      ...userInfo,
+    }
+  }, [appId, language, stage, userInfo])
```

Comment:
user info is an object, `useMemo` only compares the reference which would result in your memoized object being recreated on every component change

Reply:
looks okeyish